portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,8 +41,14 @@ public void init() {
      private String convertToJson(ArrayList<String> messages) {
         String json = "{";
          for(int i = 0 ; i< messages.size(); i++){
+            if(i == messages.size()-1){
+                json+= "\"message_"+ Integer.toString(i)+"\": ";
```

Comment:
I'm wondering how you're planning on parsing this in the JavaScript side? If you have 100 messages, how will your JavaScript know how many to expect?

What I'm getting at is, you might want to return a JSON array instead of a JSON object?